Src/BootCamp.Chapter/Shop.cs

```diff
@@ -40,6 +42,18 @@ public void Add(Item item)
         /// <param name="name"></param>
         public void Remove(string name)
         {
+            if(name is string)
```

Comment:
This is confusing to read. What you are saying is- `if it's not null, do this`. Would be more readable if you were more explicit about it. Also, do validation first and the rest of the logic after.